kotlinx-coroutines-core/jvm/test/MutexStressTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.internal.*
+import kotlinx.coroutines.selects.*
+import kotlinx.coroutines.sync.*
+import org.junit.*
+import java.util.concurrent.*
+
+class MutexStressTest : TestBase() {
+    @Test
+    fun testStressCancellationDoesNotBreakMutex() = runTest {
```

Comment:
This test fails on a regular basis both locally and on TC: https://teamcity.jetbrains.com/viewLog.html?buildId=3801302&buildTypeId=KotlinTools_KotlinxCoroutines_BuildAggregated

Reply:
Works fine both locally and on Teamcity now